Document/0x05b-Basic-Security_Testing.md

```diff
@@ -1244,10 +1244,23 @@ For enterprise tools, see the section "Static Source Code Analysis" in the chapt
 
 #### Dynamic Analysis
 
--- ToDo: <https://github.com/OWASP/owasp-mstg/issues/1240>
+Dynamic Analysis assesses the application after compiling it and running it, in a similar fashion to black box application testing. The way the code works should be inferred and looked at from a higher level. This allows the tester to properly identify business logic flaws and it helps identify vulnerabilities in live environments. The tester in dynamic analysis can either be an automated tool, such as [MobSF](https://github.com/MobSF/Mobile-Security-Framework-MobSF/), or a real penetration tester. On Android, dynamic analysis can be done by side-loading applications, or by leveraging frameworks such as [Magisk](https://github.com/topjohnwu/Magisk) or [Xposed](https://forum.xda-developers.com/xposed).
+
+_Note:_ In order to use those frameworks, a rooted phone is required.
 
 ##### Using Non-Rooted Devices
 
+Rooted devices are certainly easier to bypass security than non-rooted devices. There are still ways to analyse android devices without rooting them. This makes it more intuitive for testers to test their applications without the need of having an extra rooted phone.
+
+One clear differencing version in Android must be discussed before going forward. As of Android 8.0 (API level 26), using a non-rooted device doesn't allow the tester to bypass `SSL pinning` by injecting the proxy certificate (_e.g._ Burp Suite's certificate) as applications need to be explicitly told to trust User CAs, and only trust System CAs by default.
```

Comment:
isn't there another way to bypass that?

Reply:
This is already described here: https://github.com/OWASP/owasp-mstg/blob/dcda81fa5974146cb6efc3f61be4b12d62d8bcc6/Document/0x05b-Basic-Security_Testing.md#bypassing-the-network-security-configuration

Bypassing SSL Pinning should also be moved to the test case and not to the basic testing: https://github.com/OWASP/owasp-mstg/blob/658b35132c033aede9a260a7a7ea7e78bbf32868/Document/0x05g-Testing-Network-Communication.md#testing-custom-certificate-stores-and-certificate-pinning

Reply:
Good to hear that. I'll remove them and focus more on the need, pros and cons of non-rooted devices